### PR TITLE
Fix Conditional Rendering of TextAreaFormField for Sample Test Type

### LINKED ISSUE #8958: Text Area Not Visible When 'Other Type' is Selected in Sample Test Type
**Bug Description**  
The text area for providing details is not visible when "Other Type" is selected in the Sample Test Type.

**To Reproduce**  
Follow these steps to replicate the issue:

1. Navigate to '/patients'.
2. Click on any live patient with a consultation.
3. Open the Patient Details section.
4. Scroll down and select "Requested Sample Test".
5. In the Sample Test Type dropdown, choose the last option, "Other Type".
6. Observe the error.

**Expected Behavior**  
The text field for additional details should appear upon selecting "Other Type".

**Screenshots**  
Image

**Desktop Information:**  

- **OS:** Windows  
- **Browser:** Chrome  
- **Version:** 130.0.6723.70

### PR DESCRIPTION
**Pull request by JavidSumra**
## Proposed Changes

- Fixes #8958 
- Updated the condition to render the `TextAreaFormField` on "OTHER TYPE" selections in the sample type dropdown.
- Ensured that the component renders correctly based on the selected sample type.

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Release Notes

- **New Features**
	- Introduced internationalization support for the SampleDetails component, allowing dynamic translation of text for enhanced accessibility.
	- Expanded localization vocabulary with new medical terminology in the English language files.

- **Bug Fixes**
	- Adjusted conditional rendering for the SampleTest component to improve display logic based on sample type selection.
	- Simplified sample type display logic in the SampleTestCard component for consistency.

These updates enhance user experience by improving language support and refining component functionality.

### CODE REVIEW

**Review by rithviknishad — CHANGES_REQUESTED** [review]
solves the mentioned issue; but in the sample details page, the sample test type details (value entered in the text area) is not being shown. let's fix that too

image


image

**Comment by JavidSumra** [comment]
@rithviknishad Can I add extra space after type for that or just display it inline?

**Comment by rithviknishad** [comment]
you can add it to the right (2nd col. of the grid) like others

**Comment by JavidSumra** [comment]
@rithviknishad requested changes done

File: src/components/Patient/SampleDetails.tsx [hunk 1]
```diff
@@ -25,6 +25,7 @@ export const SampleDetails = ({ id }: DetailRoute) => {
         if (!(res?.ok && data)) {
           navigate("/not-found");
         }
+        console.log(data);
```
Comment (Jacobjeevan) [inline src/components/Patient/SampleDetails.tsx]: Let's remove this console.log 👍

File: src/components/Patient/SampleDetails.tsx [hunk 2]
```diff
@@ -411,13 +412,21 @@ export const SampleDetails = ({ id }: DetailRoute) => {
...
               <span className="font-semibold capitalize leading-relaxed">
                 Sample Type:{" "}
               </span>
               {_.startCase(_.camelCase(sampleDetails.sample_type))}
             </div>
           )}
+          {sampleDetails?.sample_type === "OTHER TYPE" && (
+            <div>
+              <span className="font-semibold capitalize leading-relaxed">
+                Sample Type Description:
```
Comment (Jacobjeevan) [inline src/components/Patient/SampleDetails.tsx]: Add a space after the colon, similar to what's there for other titles.

On a side note, now's probably the time to add in translations for these titles.

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

File: src/components/Patient/SampleDetails.tsx (hunk 2 above)
Comment (JavidSumra) [reply src/components/Patient/SampleDetails.tsx]: So shall I need to add translation for each field title?
@Jacobjeevan

**Review by khavinshankar — APPROVED** [review]

**Comment by khavinshankar** [comment]
@Jacobjeevan @rithviknishad can u please re-review this

**Review by rithviknishad — APPROVED** [review]

**Review by Jacobjeevan — APPROVED** [review]

**Comment by nihal467** [comment]
@gigincg There is no product requirement for adding a text box for selecting other sample test types in the sample test request form. Can you provide insight into whether it's a required change or not?

**Comment by JavidSumra** [comment]
Hey @nihal467 I'm not adding textbox it's already present but with the wrong condition for rendering.

**Comment by nihal467** [comment]
@JavidSumra 
image

In the Sample Test History section on the Patient Details page, the "Other Type" sample request is currently displaying the Sample Test Type Details description under the Sample Test heading. Please update it to display "Other Type" instead.

**Comment by nihal467** [comment]
LGTM

**Review by rithviknishad — APPROVED** [review]

**Pull request merged**